File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    height = len(triangle)

    # Loop from the second last row to the top
    for row in range(height - 2, -1, -1):
        for col in range(len(triangle[row])):
            # Calculate the maximum sum path for each position
            triangle[row][col] += max(triangle[row + 1][col], triangle[row + 1][col + 1])

    return triangle[0][0]

File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> triangle) {
    for (int y = triangle.size() - 2; y >= 0; --y) {
        for (int x = 0; x < triangle[y].size(); ++x) {
            triangle[y][x] += max(triangle[y+1][x], triangle[y+1][x+1]);
        }
    }
    return triangle[0][0];
}
